common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoScorer.java

```diff
@@ -373,13 +374,31 @@ private static MojoPipeline loadMojoPipelineFromFile() {
       throw new RuntimeException("Could not load mojo from file: " + mojoFile);
     }
     try {
-      MojoPipeline mojoPipeline = MojoPipelineService.loadPipeline(mojoFile);
+      MojoPipeline mojoPipeline = loadMojoPipelineFromFile(mojoFile);
       log.info("Mojo pipeline successfully loaded ({}).", mojoPipeline.getUuid());
       return mojoPipeline;
     } catch (IOException e) {
       throw new RuntimeException("Unable to load mojo from " + mojoFile, e);
     } catch (LicenseException e) {
       throw new RuntimeException("License file not found", e);
+    } catch (Exception e) {
+      throw new RuntimeException("Unexpected exception occurs", e);
     }
   }
+
+  private static MojoPipeline loadMojoPipelineFromFile(File mojoFile)
+      throws IOException, LicenseException {
+    try {
+      return MojoPipelineService.loadPipeline(mojoFile, buildPipelineConfig());
+    } catch (IllegalArgumentException e) {
```

Comment:
Instead of starting down this try-catch, **_and then_** checking if some env/property was set, it is recommended to instead explicitly first checking the env/property.

If not set, the flow would be simplified (no need for an additional try/catch)

If set, then you could be very explicit and fail without fallback if the MOJO2 runtime does not support the interval, rather than falling back.  (You can still fallback if that's a requirement, but doing so would no longer be coincidental).

Reply:
@orendain the caveat is `MojoPipelineService` will explicit the exception if `prediction_interval` is unsupported for certain model and scorer does not able to do some pre-checking to find out if the model is supported or not

Reply:
What the first part of my comment was getting at was that you could skip a lot of work in the case that the user does not want to prediction intervals.

E.g.:
```java
if ( noSpecialEnvVar ) {
  // Nothing special.
  return MojoPipelineService.loadPipeline();
}

// Only now must we do something new ...
```

In the first check, there is no `PipelineConfig` builder, no try/catch.  No introducing a different workflow than what already exists - a flow as straight as possible with minimal new places to look should something go wrong.

Reply:
we could have three `loadPipeline()` in your flow:
```
if (noSpecialEnvVar) {
  return loadPipeline(file);
}

try {
  return loadPipeline(file, prediction_interval_enabled);
} catch (IllegalArgumentException e) {
  return loadPipelien(file);
}
```
But in my approach, only two is required, am i right ?

Reply:
Correct, two paths in your example would be possible, but consider that this can be simplified down to one with some additional benefits below.

---

If there's no check for the envVar, then there's no guarantee (at least according to docs) that an `IllegalArgumentException` means simply that the user asked for prediction interval but the model does not support it.

In other words, the code will try load the pipeline, and if it fails for whatever reason due to catching an `IllegalArg...`, it will just try again without a config, regardless of if the config was the issue or not.

So ...

```java
try {
  return loadPipeline(file, prediction_interval_enabled);
} catch (IllegalArgumentException e) {
  return loadPipeline(file);
}
```

... will call `loadPipeline()` twice on _any_ `IllegalArgumentException`.  If the second `loadPipeline()` fails for any reason, then the second exception will propagate outward.

That means that if the issue was indeed that the envVar config was unsupported, that won't be the exception that makes it down the stack.  The only indicator of the config causing an issue is a debug message.

---

Consider if instead the code was:

```java
if (noSpecialEnvVar) {
  return loadPipeline(file);
}

return loadPipeline(file, prediction_interval_enabled);
```

or even simply:

```java
return loadPipeline(file, prediction_interval_enabled___OR___disabled);
```

... Then there would be a single path and the return exception would always be relevant.  The side effect either of the two above don't have is that a fallback is attempted.

Which prompts the question:  If the user has asked for a prediction range, and the scorer can't deliver it, are we sure the right move is to fallback to simply ignoring their configuration?  Or should we fail-fast?

Afterall, the new config is "PREDICTION_INTERVAL_ENABLED" and not something like "PREDICTION_INTERVAL_PREFERRED".